hack/codegen/proto.sh

```diff
@@ -105,6 +105,42 @@ function main() {
     --proto-import="${tmp_dir}/vendor" \
     --output-dir="${tmp_dir}/src"
 
+
+{ msg "Generating API docs"; } 2> /dev/null
+
+# Protoc-gen-doc plugin is used to generate API documentation.
+# Kube and Kargo API definitions are consolidated into a single directory
+# along with the template file. This temporary doc directory is
+# used as the basis for relative input paths for the proto compiler.
+# Final result is output in docs/docs/90-api-documenation.md.
+
+temp_doc_dir=$(mktemp -d)
+cp -R "${proj_dir}/api" $temp_doc_dir/api
+cp -R "${tmp_dir}/vendor/k8s.io" $temp_doc_dir/k8s.io
+cp "${proj_dir}/api-docs.templ" $temp_doc_dir/api-docs.templ
+rm "${proj_dir}/docs/docs/90-api-documentation.md" || true
+
+protoc -I $temp_doc_dir \
+	--doc_out="${proj_dir}/docs/docs" \
+	--doc_opt=$temp_doc_dir/api-docs.templ,90-api-documentation.md:k8s.io \
+    "api/service/v1alpha1/service.proto" \
+    "api/rbac/v1alpha1/generated.proto" \
+    "api/v1alpha1/generated.proto" \
+    "api/stubs/rollouts/v1alpha1/generated.proto" \
+    "k8s.io/api/core/v1/generated.proto" \
+    "k8s.io/api/batch/v1/generated.proto" \
+    "k8s.io/api/rbac/v1/generated.proto" \
+    "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/generated.proto" \
+    "k8s.io/apimachinery/pkg/util/intstr/generated.proto" \
+    "k8s.io/apimachinery/pkg/api/resource/generated.proto" \
+    "k8s.io/apimachinery/pkg/runtime/schema/generated.proto" \
+    "k8s.io/apimachinery/pkg/runtime/generated.proto" \
+    "k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto"
```

Comment:
Help me understand how this works...

I notice that all the k8s.io APIs are not included in the generated docs. I think that's the right thing to do, but I'm wondering what actually prevents them from appearing there. Is it the `:k8s.io` at the end of line +125?

The reason I'm asking is that I think we should consider `api/stubs/rollouts/v1alpha1/generated.proto` _also_ not being included in the final product.

These docs already feel a bit like drinking from a fire hose, so if we can improve the signal:noise ratio by omitting things where we can reasonably claim they're documented better elsewhere, I think we ought to do so.